src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSOutputFormatProvider.java

```diff
@@ -0,0 +1,268 @@
+
+    @Override
+    public void commitJob(JobContext jobContext) throws IOException {
+      delegate.commitJob(jobContext);
+    }
+
+    @Override
+    public void abortJob(JobContext jobContext, JobStatus.State state) throws IOException {
+      delegate.abortJob(jobContext, state);
+    }
+
+    @Override
+    public void setupTask(TaskAttemptContext taskAttemptContext) throws IOException {
+      delegate.setupTask(taskAttemptContext);
+    }
+
+    @Override
+    public boolean needsTaskCommit(TaskAttemptContext taskAttemptContext) throws IOException {
+      return delegate.needsTaskCommit(taskAttemptContext);
+    }
+
+    @Override
+    public void commitTask(TaskAttemptContext taskAttemptContext) throws IOException {
+      delegate.commitTask(taskAttemptContext);
+      //Once the task is committed , get the committed path and update the count metadata on file.
+      try {
+        updateMetricMetaData(taskAttemptContext);
+      } catch (IOException exception) {
+        LOG.warn("Unable to record metric for task. Metric emitted for the number of affected rows may be incorrect.",
+                 exception);
+      }
+    }
+
+    private void updateMetricMetaData(TaskAttemptContext taskAttemptContext) throws IOException {
+      if (!(delegate instanceof FileOutputCommitter)) {
+        return;
+      }
+
+      FileOutputCommitter fileOutputCommitter = (FileOutputCommitter) delegate;
+      Configuration configuration = taskAttemptContext.getConfiguration();
+      Path committedTaskPath = fileOutputCommitter.getCommittedTaskPath(taskAttemptContext);
+
+      if (configuration == null || committedTaskPath == null) {
+        return;
+      }
+
+      try (FileSystem fileSystem = committedTaskPath.getFileSystem(configuration)) {
+        if (!(fileSystem instanceof GoogleHadoopFileSystem)) {
+          return;
+        }
+        updateInGCSFS(taskAttemptContext, configuration, committedTaskPath,
+                      (GoogleHadoopFileSystem) fileSystem);
+
+      }
+    }
+
+    private void updateInGCSFS(TaskAttemptContext taskAttemptContext, Configuration configuration,
+                               Path committedTaskPath, GoogleHadoopFileSystem fileSystem) throws IOException {
+      GoogleCloudStorageFileSystem gcsFs = fileSystem.getGcsFs();
+      if (gcsFs == null) {
+        return;
+      }
+
+      try {
+        List<URI> files = gcsFs.listFileNames(gcsFs.getFileInfo(committedTaskPath.toUri()));
+        if (files == null || files.isEmpty()) {
+          return;
+        }
+        //update metadata on any file - typically only one will be there
+        GCSPath objPath = GCSPath.from(files.get(0).toString());
+        //read the count from configuration
+        String keyInConfig = String.format(RECORD_COUNT_FORMAT, taskAttemptContext.getTaskAttemptID());
+        Map<String, byte[]> metaData = new HashMap<>();
+        byte[] bytes = Long.valueOf(configuration.getLong(keyInConfig, 0)).toString().getBytes();
```

Comment:
Provide charset. Also simpler to use `String.valueOf(conf.getLong(keyInConfig, 0L))`

Reply:
Done